packages/core/saved-objects/core-saved-objects-server/src/extensions/security.ts

```diff
@@ -112,94 +55,262 @@ export interface CheckAuthorizationResult<A extends string> {
 }
 
 /**
- * The EnforceAuthorizationParams interface contains settings for
- * enforcing a single action via the ISavedObjectsSecurityExtension.
+ * The AuthorizeObject interface contains information to specify an
+ * object for authorization. This is a base interface which is
+ * extended by other interfaces for specific actions.
  */
-export interface EnforceAuthorizationParams<A extends string> {
-  /**
-   * A map of types to spaces that will be affected by the action
-   */
-  typesAndSpaces: Map<string, Set<string>>;
+export interface AuthorizeObject {
+  /** The type of object */
+  type: string;
+  /** The id of the object */
+  id: string;
+}
+
+/**
+ * The AuthorizeObjectWithExistingSpaces extends AuthorizeObject and contains
+ * an array of existing namespaces for the object. Used by the
+ * authorizeDelete, authorizeBulkDelete, authorizeGet,
+ * authorizeCheckConflicts, and getFindRedactTypeMap methods.
+ */
+export interface AuthorizeObjectWithExistingSpaces extends AuthorizeObject {
   /**
-   * The relevant action (create, update, etc.)
+   * Spaces where the object is known to exist. Usually populated
+   * by document data from the result of an es query.
    */
-  action: A;
+  existingNamespaces?: string[];
```

Comment:
why would an object not have `existingNamespaces` in terms of the API it doesn't feel like `AuthorizeObjectWithExistingSpaces` should be able to say "I don't know what spaces I'm in?"